langchain/schema.py

```diff
@@ -17,6 +19,15 @@
 
 from pydantic import BaseModel, Extra, Field, root_validator
 
+if TYPE_CHECKING:
```

Comment:
This lets us define here without a cyclic dependency but if we're breaking enough things then may as well do another big push

Reply:
The solution here is to split schema into multiple files like we've done in js